base/server/python/pki/server/upgrade.py

```diff
@@ -86,6 +90,10 @@ def copydirs(self, source, dest, force=False):
     def copyfile(self, source, dest, force=False):
         self.instance.copyfile(source, dest, force=force)
 
+    def init_scriptlet(self, scriptlet):
```

Comment:
Maybe I'm missing something obvious. This new method never seems to be invoked?

Reply:
This method is used in the superclass to initialize the scriptlet object before running it:
https://github.com/dogtagpki/pki/blob/master/base/common/python/pki/upgrade.py#L458

The code above overrides the method to insert the instance object into the scriptlet
object, then call the original method.